sql/driver/conn.go

```diff
@@ -178,13 +178,18 @@ func (c *conn) Send(call sqlwire.Call) (*rows, error) {
 			} else if datum.IntVal != nil {
 				t[j] = *datum.IntVal
 			} else if datum.UintVal != nil {
-				t[j] = *datum.UintVal
+				// uint64 not supported by the driver.Value interface.
+				t[j] = int64(*datum.UintVal)
 			} else if datum.FloatVal != nil {
 				t[j] = *datum.FloatVal
 			} else if datum.BytesVal != nil {
 				t[j] = datum.BytesVal
 			} else if datum.StringVal != nil {
-				t[j] = *datum.StringVal
+				t[j] = []byte(*datum.StringVal)
+			}
+			if !driver.IsScanValue(t[j]) {
+				err := fmt.Errorf("Value of bad type %T returned by database", t[j])
```

Comment:
okay, converted to panic

On Wed, Jul 1, 2015 at 3:49 PM Peter Mattis <notifications@github.com>
wrote:

> In sql/driver/conn.go
> <https://github.com/cockroachdb/cockroach/pull/1604#discussion_r33716952>:
>
> >  			} else if datum.FloatVal != nil {
> >  				t[j] = *datum.FloatVal
> >  			} else if datum.BytesVal != nil {
> >  				t[j] = datum.BytesVal
> >  			} else if datum.StringVal != nil {
> > -				t[j] = *datum.StringVal
> > +				t[j] = []byte(*datum.StringVal)
> > +			}
> > +			if !driver.IsScanValue(t[j]) {
> > +				err := fmt.Errorf("Value of bad type %T returned by database", t[j])
>
> t[j] being unassigned will be nil and will pass the driver.IsScanValue()
> test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1604/files#r33716952>.
>